### ISSUE TITLE
Reduce memory usage

### ISSUE THREAD

**Issue by nayfaan** [nayfaan/puni_searching#2]
```[tasklist]
### Reorganizing items
- [ ] Items should be group by categories
-- [ ] Categories should be serialized
-- [ ] Items should be restructured into dictionary sorted by category
- [ ] Less information should be sent into itertoolsCombinations and zip functions
```

**Comment by nayfaan** [comment]
c512541

```[tasklist]
### Reorganizing items
- [ ] Items should be group by categories
-- [✔︎] Categories should be serialized
-- [ ] Items should be restructured into dictionary sorted by category
- [ ] Less information should be sent into itertoolsCombinations and zip functions
```

**Comment by nayfaan** [comment]
434dff4

```[tasklist]
### Reorganizing items
- [ ] Items should be group by categories
-- [✔︎] Categories should be serialized
-- [ ] Items should be restructured into dictionary sorted by category
- [✔︎] Less information should be sent into itertoolsCombinations and zip functions
```

**Comment by nayfaan** [comment]
fixed in 60ae900
